components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/impl/UserIdApiServiceImpl.java

```diff
@@ -63,9 +72,33 @@ public Response terminateSessionBySessionId(String userId, String sessionId) {
     @Override
     public Response terminateSessionsByUserId(String userId) {
 
-        Util.validateUserId(SessionManagementServiceHolder.getRealmService(), userId,
-                ContextLoader.getTenantDomainFromContext());
-        sessionManagementService.terminateSessionsByUserId(userId);
-        return Response.noContent().build();
+        try {
+            UserRealm userRealm = CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+
+            if (userStoreManager == null) {
+                log.debug("Returns Null by UserStore Manager");
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+            }
+
+            String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            String adminUserName = userRealm.getRealmConfiguration().getAdminUserName();
+            String adminUserID = userStoreManager.getUserIDFromUserName(adminUserName);
+
+            if (!StringUtils.equals(username, adminUserName) && StringUtils.equals(userId, adminUserID)) {
+                log.debug("Forbidden operation Collaborator having admin permissions is trying to" +
+                        " terminate the organization owner's sessions ");
+                return Response.status(Response.Status.FORBIDDEN).build();
+            }
+
+            Util.validateUserId(SessionManagementServiceHolder.getRealmService(), userId,
+                    ContextLoader.getTenantDomainFromContext());
+            sessionManagementService.terminateSessionsByUserId(userId);
+            return Response.noContent().build();
+        } catch (UserStoreException e) {
+            log.error("Error occurred while invoking user UserStore.", e);
```

Comment:
invoking userstore manager

Reply:
Update in commit a896550c81e986cd0d2e3edbc127b45276d4b491

Reply:
Update in commit 60060472ecf3b2684787602942d489bbaa11a338